packages/react-dom/src/__tests__/ReactCompositeComponent-test.js

```diff
@@ -326,6 +326,32 @@ describe('ReactCompositeComponent', () => {
     expect(cbCalled).toBe(false);
   });
 
+  it(
+    'should throw an Error with a warning when rendering' +
+      "a class with a render method that doesn't extend React.Component",
+    () => {
+      spyOn(console, 'error');
+      var container = document.createElement('div');
+      class ClassWithRenderNotExtended {
+        render() {
+          return <div />;
+        }
+      }
+      expectDev(console.error.calls.count()).toBe(0);
+      try {
+        ReactDOM.render(<ClassWithRenderNotExtended />, container);
+      } catch (e) {
+        expect(e).toEqual(new TypeError('Cannot call a class as a function'));
```

Comment:
This only runs assertions if we get into `catch`. But what if we have a bug that causes us *not* to throw an error? Then the test doesn't assert anything.